packages/headless/src/database/mongodb/store-admin-role/store-admin.role.model.ts

```diff
@@ -0,0 +1,43 @@
+import { Role } from 'src/entity/role';
+import { Schema, model } from 'mongoose';
+import { randomUUID } from 'crypto';
+import { RoleTypeEnum } from 'models';
+
+const StoreAdminRoleSchema = new Schema<Role>(
+  {
+    id: {
+      type: String,
+      default: () => randomUUID(),
+      unique: true,
+      indexe: true,
+    },
+    name: {
+      type: String,
+      trim: true,
+      required: true,
+    },
+    roleType:{
+      type: String,
+      index: true,
+      default: RoleTypeEnum.STORE_ADMIN
```

Comment:
using this default value may cause unexpected scenario in some case. For example, if in an branch admin create API, the developer forgot to set this value, then what will happen? The new admin will become the Store Admin!!
So either remove this default value, or set this value to be BRANCH_ADMIN